src/components/notebook-utils.js

```diff
@@ -48,8 +48,18 @@ export const notebookNameValidator = existing => ({
     message: 'already exists'
   }
 })
+// removes leading dirs and a file ext suffix on paths
+export const getDisplayName = name => stripExtension(getFileName(name))
 
-export const notebookNameInput = ({ inputProps, ...props }) => h(ValidatedInput, {
+// removes all paths up to and including the last slash
+export const getFileName = _.flow(_.split('/'), _.last)
+
+// removes a file extension from a string
+export const stripExtension = _.replace(/\.[^/.]+$/, '')
```

Comment:
If you define this after `getExtension`, it can be:

```js
const stripExtension = str => str.slice(0, -(getExtension(str).length + 1)) // + 1 to account for the period
```